backend/src/logic/svgRenderer.ts

```diff
@@ -110,13 +118,17 @@ export function svgRenderer(options: RendererOptions): {
       lines.forEach((line, index) => {
         if (line.length === 0) {
           if (lineNumber) {
-            svg += `<text font-family="${fontFamily}" font-size="${fontSize}" y="${lineHeight * (index + 1)}">\n`;
+            svg += `<text font-family="${fontFamily}" font-size="${fontSize}" y="${
+              lineHeight * (index + 1 + offsetY)
+            }">\n`;
             svg += generateLineNumber(index, { fontFamily, fontWidth, lineHeight, fg });
             svg += '</text>';
           }
           svg += `\n`;
         } else {
-          svg += `<text font-family="${fontFamily}" font-size="${fontSize}" y="${lineHeight * (index + 1)}">\n`;
+          svg += `<text font-family="${fontFamily}" font-size="${fontSize}" y="${
+            lineHeight * (index + 1 + offsetY)
+          }">\n`;
           if (lineNumber) svg += generateLineNumber(index, { fontFamily, fontWidth, lineHeight, fg });
```

Comment:
might as well do something like: 

```javascript
lines.forEach((line, index) => {
    const idx = index + 1;
```

and make `generateLineNumber` accepts 1-based index instead of zero so we don't have to keep doing `index + 1` on different places (e.g `idx + offsetY` instead of `index + 1 + offsetY`)